From: Ben Hutchings <ben@decadent.org.uk>
Date: Thu, 10 Nov 2011 03:46:57 +0000
Subject: [PATCH] time: Avoid ABI change in 2.6.32.47

Change jiffies_to_clock_t() parameter type back to long and convert
to unsigned long inside the function.
---
 include/linux/jiffies.h |    2 +-
 kernel/time.c           |    3 ++-
 2 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/include/linux/jiffies.h b/include/linux/jiffies.h
index fbd9836..1a9cf78bf 100644
--- a/include/linux/jiffies.h
+++ b/include/linux/jiffies.h
@@ -303,7 +303,7 @@ extern void jiffies_to_timespec(const unsigned long jiffies,
 extern unsigned long timeval_to_jiffies(const struct timeval *value);
 extern void jiffies_to_timeval(const unsigned long jiffies,
 			       struct timeval *value);
-extern clock_t jiffies_to_clock_t(unsigned long x);
+extern clock_t jiffies_to_clock_t(long x);
 extern unsigned long clock_t_to_jiffies(unsigned long x);
 extern u64 jiffies_64_to_clock_t(u64 x);
 extern u64 nsec_to_clock_t(u64 x);
diff --git a/kernel/time.c b/kernel/time.c
index 33df60e..470a768 100644
--- a/kernel/time.c
+++ b/kernel/time.c
@@ -593,8 +593,9 @@ EXPORT_SYMBOL(jiffies_to_timeval);
 /*
  * Convert jiffies/jiffies_64 to clock_t and back.
  */
-clock_t jiffies_to_clock_t(unsigned long x)
+clock_t jiffies_to_clock_t(long x0)
 {
+	unsigned long x = x0;
 #if (TICK_NSEC % (NSEC_PER_SEC / USER_HZ)) == 0
 # if HZ < USER_HZ
 	return x * (USER_HZ / HZ);
-- 
1.7.7

